Surface_mesh_topology/doc/Surface_mesh_topology/Surface_mesh_topology.txt

```diff
@@ -262,9 +285,51 @@ The canonical form of a curve is obtained by flattening its brackets, removing i
+\cgalFigureBegin{fig_switch_sample,switch_sample.svg}
+ Top-left, an intersection between red subpath and green subpath at the start of a left L-shaped red subpath. Top-right, switch the left L-shape to right L-shape to avoid the intersection. Bottom, no switch because no intersection can be avoided.
+    \cgalFigureEnd
+
+\subsubsection SMTopology_Simplicity_Test_Relative_Order Decide Relative Order
+Since the algorithm inductively builds an ordering, it has to determine relative order between the edge being processed and edges that has been ordered. There are three cases to consider.
+
+-# The predecessor of the current edge is adjacent to another edge which forms the same turn of the predecessor and the current edge. In this case, the current edge must be adjacent to the edge forming the turn. See \cgalFigureRef{fig_relative_order_corner}.
+\cgalFigureBegin{fig_relative_order_corner,relative_order_corner.svg}
+ The red edge is being processed. The blue edge is adjacent to green edge (the predecessor of the red edge) in the previously constructed ordering and forms a turn (blue-pink turn) same to the green-red turn. The red edge must be right next to the pink edge in the ordering.
+    \cgalFigureEnd
+
+-# Comparing the current edge with another edge by the predecessor of the current edge. If the edge being compared against has predecessor/successor extended to the direction of the predecessor of the current edge, and they doesn't form the same turn. The relative order of edges around the vertex can be used as the relative order between the edges. See \cgalFigureRef{fig_relative_order_normal}.
+\cgalFigureBegin{fig_relative_order_normal,relative_order_normal.svg}
+ The red edge is being processed and is compared against the pink edge. Since the green edge (the predecessor of the red edge) is to the right of the blue edge around the vertex, red edge must be to the right of the pink edge in the ordering.
+    \cgalFigureEnd
+
+-# If the current edge crosses the same edge in the quadrangulation as the first edge of the path, the predecessor of the first edge (aka the last edge of the path) has not been processed yet. Hence if they form the same turn, we cannot determine the relative order of the predecessors. So the idea is to keep following the predecessors of current edge and the first edge until they diverge, at which point the relative order around the vertex can be used to determine the relative order. See \cgalFigureRef{fig_relative_order_first}. This can be precomputed by finding the all the longest common suffices of the path against its circular shifts. A modified Knuth-Morris-Pratt algorithm is applied to preprocess the path in linear time.
```

Comment:
```suggestion
-# If the current edge crosses the same edge in the quadrangulation as the first edge of the path, the predecessor of the first edge (aka the last edge of the path) has not been processed yet. Hence if they form the same turn, we cannot determine the relative order of the predecessors. So the idea is to keep following the predecessors of current edge and the first edge until they diverge, at which point the relative order around the vertex can be used to determine the relative order. See \cgalFigureRef{fig_relative_order_first}. This can be precomputed by finding all the longest common suffices of the path against its circular shifts. A modified Knuth-Morris-Pratt algorithm is applied to preprocess the path in linear time.
```